File: backend/schemas/payment.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from models.transaction import TransactionStatus


class UserSummary(BaseModel):
    id: uuid.UUID
    full_name: str
    email: str
    wallet_address: str | None

    model_config = {"from_attributes": True}


class TransactionResponse(BaseModel):
    id: uuid.UUID
    invoice_id: uuid.UUID
    payer_id: uuid.UUID
    payee_id: uuid.UUID
    amount: float
    currency: str
    status: TransactionStatus
    circle_transfer_id: str | None
    tx_hash: str | None
    payer_wallet: str | None
    payee_wallet: str | None
    ai_receipt: str | None
    error_message: str | None
    created_at: datetime
    confirmed_at: datetime | None

    payer: UserSummary
    payee: UserSummary

    model_config = {"from_attributes": True}


class PaymentInitiateResponse(BaseModel):
    transaction: TransactionResponse
    message: str

File: frontend/types/payment.ts
export enum TransactionStatus {
  PENDING = "PENDING",
  CONFIRMED = "CONFIRMED",
  FAILED = "FAILED",
}

export interface TransactionUser {
  id: string;
  full_name: string;
  email: string;
  wallet_address: string | null;
}

export interface Transaction {
  id: string;
  invoice_id: string;
  payer_id: string;
  payee_id: string;
  amount: number;
  currency: string;
  status: TransactionStatus;
  circle_transfer_id: string | null;
  tx_hash: string | null;
  payer_wallet: string | null;
  payee_wallet: string | null;
  ai_receipt: string | null;
  error_message: string | null;
  created_at: string;
  confirmed_at: string | null;
  payer: TransactionUser;
  payee: TransactionUser;
}

export interface PaymentInitiateResponse {
  transaction: Transaction;
  message: string;
}
